go/mysql/conn_params.go

```diff
@@ -16,19 +16,24 @@ limitations under the License.
 
 package mysql
 
-import "vitess.io/vitess/go/vt/vttls"
+import (
+	"vitess.io/vitess/go/mysql/collations"
+	"vitess.io/vitess/go/vt/vttls"
+)
 
 // ConnParams contains all the parameters to use to connect to mysql.
 type ConnParams struct {
-	Host       string `json:"host"`
-	Port       int    `json:"port"`
-	Uname      string `json:"uname"`
-	Pass       string `json:"pass"`
-	DbName     string `json:"dbname"`
-	UnixSocket string `json:"unix_socket"`
-	Charset    string `json:"charset"`
-	Flags      uint64 `json:"flags"`
-	Flavor     string `json:"flavor,omitempty"`
+	Host                 string                  `json:"host"`
+	Port                 int                     `json:"port"`
+	Uname                string                  `json:"uname"`
+	Pass                 string                  `json:"pass"`
+	DbName               string                  `json:"dbname"`
+	UnixSocket           string                  `json:"unix_socket"`
+	Charset              string                  `json:"charset"`
+	Collation            string                  `json:"collation"`
+	Flags                uint64                  `json:"flags"`
+	Flavor               string                  `json:"flavor,omitempty"`
+	CollationEnvironment *collations.Environment `json:"collation_environment,omitempty"`
```

Comment:
I am not sure this is the right place to store a collation environment. Among other semantic questions, the environment cannot be serialized as JSON at all so...

Reply:
After pairing up, we decided to move `CollationEnvironment` to the `mysql.Conn` instead.